stl/inc/xmemory

```diff
@@ -460,6 +460,15 @@ template <class _Ty>
 struct _Is_default_allocator<allocator<_Ty>, void_t<typename allocator<_Ty>::_From_primary>>
     : is_same<typename allocator<_Ty>::_From_primary, allocator<_Ty>>::type {};
 
+#if _HAS_CXX23
+template <class _Alloc, class _SizeTy, class = void>
+inline constexpr bool _Has_member_allocate_at_least = false;
+
+template <class _Alloc, class _SizeTy>
+inline constexpr bool _Has_member_allocate_at_least<_Alloc, _SizeTy,
+    void_t<decltype(_STD declval<_Alloc&>().allocate_at_least(_STD declval<_SizeTy&>()))>> = true;

```

Comment:
Nitpick: WG21-N4928 [\[allocator.requirements.general\]/2.16](https://eel.is/c++draft/allocator.requirements.general#2.16) says "`n` denotes a value of type `XX::size_type`" without being specific about its cv-quals and value category.

Given that the actual call passes `const size_type _Count`, I think we should use `const _SizeTy&` here. This would also be consistent with our SFINAE for `allocate()`-with-hint, where we use `const _Size_type&`: https://github.com/microsoft/STL/blob/994f94199c561a38a4a8e15584030160095bc25a/stl/inc/xmemory#L495-L496